src/main/java/org/springframework/metrics/instrument/Measurement.java

```diff
@@ -0,0 +1,99 @@
+/**
+ * Copyright 2017 Pivotal Software, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.metrics.instrument;
+
+import java.util.*;
+
+/**
+ * A measurement sampled from a meter.
+ *
+ * @author Clint Checketts
+ * @author Jon Schneider
+ */
+public final class Measurement {
+
+    private final String name;
+    private final SortedSet<Tag> tags = new TreeSet<>(Comparator.comparing(Tag::getKey));
+    private final double    value;
```

Comment:
Why do we keep name and tags separate? Why not reuse MeterId? As a bonus the Measurement can just reuse the reference to the parent Meter's MeterId